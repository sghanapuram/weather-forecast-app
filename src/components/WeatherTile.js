import React, { Fragment } from 'react';

const WeatherTile = ({ forecastInfo }) => (
    <Fragment>
        {
            forecastInfo.map((item, i) => (
                <div key={ i } className="weather-tile">
                    <p className="city">{ item.city }</p>
                    <p className="weather-state">{ item.weatherInfo.weather_state_name }</p>
                    <img
                        src={ `https://www.metaweather.com/static/img/weather/${item.weatherInfo.weather_state_abbr}.svg` }
                        alt="weather-icon"
                        className="weather-icon"
                    />
                    <p className="current-temp">{ Math.floor(item.weatherInfo.the_temp) }&deg;</p>
                    <p className="temp">
                        H:{ Math.floor(item.weatherInfo.max_temp) }&deg;
                        <span className="min-temp">L:{ Math.floor(item.weatherInfo.min_temp) }&deg;</span>
                    </p>
                </div>
            ))
        }
    </Fragment>
);

export default WeatherTile;
