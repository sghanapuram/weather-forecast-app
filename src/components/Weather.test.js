import { render, screen } from '@testing-library/react';
import WeatherTile from './WeatherTile';

const forecastInfo = [
    {
        city: 'New York',
        woied: 1234,
        weatherInfo: {
            weather_state_name: 'Clear',
            weather_state_abbr: 'C',
            the_temp: 12.4,
            max_temp: 13.7,
            min_temp: 6
        }
    }
];

test('renders Weather Forecast Tiles ', () => {
    const container = render(<WeatherTile forecastInfo={ forecastInfo } />);
    expect(container).toMatchSnapshot();
});

test('renders Weather Forecast Tiles with city name', () => {
    render(<WeatherTile forecastInfo={ forecastInfo } />);
    const cityName = screen.getByText('New York');
    expect(cityName).toBeInTheDocument();
});

test('renders Weather Forecast Tiles with weather desc', () => {
    render(<WeatherTile forecastInfo={ forecastInfo } />);
    const weatherDesc = screen.getByText('Clear');
    expect(weatherDesc).toBeInTheDocument();
});

test('renders Weather Forecast Tiles with weather icon', () => {
    render(<WeatherTile forecastInfo={ forecastInfo } />);
    const weatherIcon = screen.getByAltText('weather-icon');
    expect(weatherIcon).toBeInTheDocument();
});

test('renders Weather Forecast Tiles with temp', () => {
    render(<WeatherTile forecastInfo={ forecastInfo } />);
    const temp = screen.getByText('12°');
    expect(temp).toBeInTheDocument();
});

test('renders Weather Forecast Tiles with min and max temp ', () => {
    render(<WeatherTile forecastInfo={ forecastInfo } />);
    const maxTemp = screen.getByText('H:13°');
    expect(maxTemp).toBeInTheDocument();
    const minTemp = screen.getByText('L:6°');
    expect(minTemp).toBeInTheDocument();
});

