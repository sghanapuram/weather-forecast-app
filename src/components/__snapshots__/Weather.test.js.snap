// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Weather Forecast Tiles  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="weather-tile"
      >
        <p
          class="city"
        >
          New York
        </p>
        <p
          class="weather-state"
        >
          Clear
        </p>
        <img
          alt="weather-icon"
          class="weather-icon"
          src="https://www.metaweather.com/static/img/weather/C.svg"
        />
        <p
          class="current-temp"
        >
          12
          °
        </p>
        <p
          class="temp"
        >
          H:
          13
          °
          <span
            class="min-temp"
          >
            L:
            6
            °
          </span>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="weather-tile"
    >
      <p
        class="city"
      >
        New York
      </p>
      <p
        class="weather-state"
      >
        Clear
      </p>
      <img
        alt="weather-icon"
        class="weather-icon"
        src="https://www.metaweather.com/static/img/weather/C.svg"
      />
      <p
        class="current-temp"
      >
        12
        °
      </p>
      <p
        class="temp"
      >
        H:
        13
        °
        <span
          class="min-temp"
        >
          L:
          6
          °
        </span>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
